lib/req/steps.ex

```diff
@@ -374,6 +374,26 @@ defmodule Req.Steps do
     end
   end
 
+  @doc """
+  Compresses the request body.
+
+  ## Request options
+
+    * `:compress_request` - compresses the request body using `gzip` compression and sets the
+    `"Content-Encoding: gzip"` header. Defaults to `false`.
+
+  """
+  @doc step: :request
+  def compress_request(request) do
+    if request.options[:compress_request] do
```

Comment:
1. let's call the option `:compress` (even if we're gonna end up renaming the step)

2. let's support other compression algorithms two (well, only one more for now):

```elixir
case Map.fetch(request.options, :compress) do
  {:ok, :gzip} ->
    request
    |> Map.update!(:body, &:zlib.gzip/1)
    |> put_header

  {:ok, :deflate} ->
    request
    |> Map.update!(:body, &:zlib.zip/1)
    |> put_header

  {:ok, algorithm} ->
    raise "expected :compress to be :gzip or :deflate, got: #{inspect(algorithm)}"

  :error ->
    request
end
```

Reply:
Please ignore this, compress: true is fine for now.

Reply:
Will do but to be honest I think clarifying what is being compressed is helpful here also.